src/groups/groups.service.ts

```diff
@@ -0,0 +1,88 @@
+import { Injectable } from '@nestjs/common';
+import { InjectModel } from '@nestjs/sequelize';
+import { Sequelize } from 'sequelize';
+
+import { CreateGroupDto } from './dto/create-group.dto';
+import { UpdateGroupDto } from './dto/update-group.dto';
+import { groups } from './entities/group.entity';
+import {
+  permissions_groups,
+  permissions,
+} from '../permissions/entities/permissions.entity';
+
+@Injectable()
+export class GroupsService {
+  constructor(
+    @InjectModel(groups)
+    private groupsModel: typeof groups,
+    @InjectModel(permissions)
+    private permissionsModel: typeof permissions,
+    @InjectModel(permissions_groups)
+    private permissionsGroupsModel: typeof permissions_groups,
+    private sequelize: Sequelize,
+  ) {}
+
+  async create(createGroupDto: CreateGroupDto) {
+    const t = await this.sequelize.transaction();
+    try {
+      const group = await groups.create(
+        {
+          name: createGroupDto.name,
+        },
+        { transaction: t },
+      );
+
+      await Promise.all(
+        createGroupDto.permission.map(async (permission) => {
+          const permissionData = await this.permissionsModel.findOne({
+            where: {
+              permission,
+            },
+            transaction: t,
+          });
+          await this.permissionsGroupsModel.create(
+            {
+              permissions_id: permissionData.getDataValue('id'),
+              group_id: group.getDataValue('id'),
+            },
+            { transaction: t },
+          );
+        }),
+      );
+      await t.commit();
+      return group;
+    } catch (error) {
+      await t.rollback();
+      console.log(error.message);
```

Comment:
I think this is not a good practice to use `console.log` in the code. You can use NestJS logger service, or do not log it at all, but passthrough an error to the response

Reply:
fixed